sycl/include/sycl/detail/pi.h

```diff
@@ -1325,9 +1326,11 @@ __SYCL_EXPORT pi_result piProgramRelease(pi_program program);
 /// \param spec_id integer ID of the constant
 /// \param spec_size size of the value
 /// \param spec_value bytes of the value
-__SYCL_EXPORT pi_result
-piextProgramSetSpecializationConstant(pi_program prog, pi_uint32 spec_id,
-                                      size_t spec_size, const void *spec_value);
+/// \param kernel is the pi_kernel to which an implicit argument for emulated
+/// specialisation constant would be added
+__SYCL_EXPORT pi_result piextProgramSetSpecializationConstant(
+    pi_program prog, pi_uint32 spec_id, size_t spec_size,
+    const void *spec_value, pi_kernel kernel);
```

Comment:
Perhaps `kernel` argument would better be positioned after `program`

Reply:
Agreed, done in: d5f876ef07aca617d1844c59a8deee8eb90b3e40